File: frontend/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  genre: string;
  rating: number;
  watched: boolean;
  created_at: string;
}

export interface MovieInput {
  title: string;
  genre: string;
  rating: number;
  watched: boolean;
}

export type WatchedFilter = "all" | "watched" | "unwatched";

File: backend/app/models/movie.py
from datetime import datetime

from sqlalchemy import Boolean, CheckConstraint, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database.session import Base


class Movie(Base):
    __tablename__ = "movies"
    __table_args__ = (CheckConstraint("rating >= 1 AND rating <= 10", name="rating_range"),)

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    genre: Mapped[str] = mapped_column(String(100), nullable=False)
    rating: Mapped[int] = mapped_column(Integer, nullable=False)
    watched: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
